src/pipecat/services/heygen/client.py

```diff
@@ -0,0 +1,613 @@
+        try:
+            await self._initialize()
+
+            self._event_queue = WatchdogQueue(self._task_manager)
+            self._event_task = self._task_manager.create_task(
+                self._callback_task_handler(self._event_queue),
+                f"{self}::event_callback_task",
+            )
+        except Exception as e:
+            logger.error(f"Failed to setup HeyGenClient: {e}")
+            await self.cleanup()
+
+    async def cleanup(self) -> None:
+        """Cleanup client resources.
+
+        Closes the active HeyGen session and resets internal state.
+        """
+        try:
+            if self._heyGen_session is not None:
+                await self._api.close_session(self._heyGen_session.session_id)
+                self._heyGen_session = None
+                self._connected = False
+
+            if self._event_task and self._task_manager:
+                self._event_queue.cancel()
+                await self._task_manager.cancel_task(self._event_task)
+                self._event_task = None
+        except Exception as e:
+            logger.exception(f"Exception during cleanup: {e}")
+
+    async def start(self, frame: StartFrame, audio_chunk_size: int) -> None:
+        """Start the client and establish all necessary connections.
+
+        Initializes WebSocket and LiveKit connections using the provided configuration.
+        Sets up audio processing with the specified sample rates.
+
+        Args:
+            frame: Initial configuration frame containing audio parameters
+            audio_chunk_size: Audio chunk size for output processing
+        """
+        if self._websocket is not None:
+            logger.debug("heygen client already started")
+            return
+
+        logger.debug(f"HeyGenClient starting")
+        self._in_sample_rate = self._params.audio_in_sample_rate or frame.audio_in_sample_rate
+        self._out_sample_rate = self._params.audio_out_sample_rate or frame.audio_out_sample_rate
+        self._send_interval = (audio_chunk_size / self._out_sample_rate) / 2
+        logger.debug(f"HeyGenClient send_interval: {self._send_interval}")
+        await self._ws_connect()
+        await self._livekit_connect()
+
+    async def stop(self) -> None:
+        """Stop the client and terminate all connections.
+
+        Disconnects from WebSocket and LiveKit endpoints, and performs cleanup.
+        """
+        logger.debug(f"HeyGenVideoService stopping")
+        await self._ws_disconnect()
+        await self._livekit_disconnect()
+        await self.cleanup()
+
+    # websocket connection methods
+    async def _ws_connect(self):
+        """Connect to HeyGen websocket endpoint."""
+        try:
+            if self._websocket:
+                logger.debug(f"HeyGenClient ws already connected!")
+                return
+            logger.debug(f"HeyGenClient ws connecting")
+            self._websocket = await websockets.connect(
```

Comment:
Required for asyncio websockets implementation:

```suggestion
            self._websocket = await websocket_connect(
```

Reply:
Done.